llvm/lib/Transforms/Scalar/LoopIdiomRecognize.cpp

```diff
@@ -1494,7 +1512,17 @@ bool LoopIdiomRecognize::runOnNoncountableLoop() {
 
   return recognizePopcount() || recognizeAndInsertFFS() ||
          recognizeShiftUntilBitTest() || recognizeShiftUntilZero() ||
-         recognizeShiftUntilLessThan();
+         recognizeShiftUntilLessThan() || recognizeAndInsertStrLen();
+}
+
+/// Check if a Value is either a nullptr or a constant int zero
+static bool isZeroConstant(const Value *Val) {
+  if (isa<ConstantPointerNull>(Val))
+    return true;
+  const ConstantInt *CmpZero = dyn_cast<ConstantInt>(Val);
+  if (!CmpZero || !CmpZero->isZero())
+    return false;
+  return true;
```

Comment:
This looks like `match(Val, m_Zero())`. Though I'm not super clear on why this patch has to add the pointer handling at all.

Reply:
Thank you for taking. You are right, it's no longer needed. We used to use this to condition the loop preheader on a `nullptr` comparison for the base of the string. I will clean this up